src/pip/_internal/models/link.py

```diff
@@ -97,7 +97,7 @@ def filename(self):
             return netloc
 
         name = urllib_parse.unquote(name)
-        assert name, ('URL %r produced no filename' % self._url)
+        assert name, ('URL {} produced no filename'.format(self._url))
```

Comment:
`!r`. I didn’t read all the diff, there’s just too much. It might be a good idea to split this into several batches.

Reply:
Yes, I will work on these and push them in batches.